src/services/data/data.service.ts

```diff
@@ -1,19 +1,43 @@
 import { DGraphService } from "../../db/dgraph.service";
 import { DataRepository } from "../data/data.repository";
-import { Hashed } from "../uprtcl/types";
+import { Hashed, Secured, Commit } from "../uprtcl/types";
+import { UprtclRepository } from "../uprtcl/uprtcl.repository";
+
+const propertyOrder = ['creatorsIds', 'dataId', 'message', 'timestamp', 'parentsIds'];
 
 export class DataService {
 
   constructor(
     protected db: DGraphService, 
-    protected dataRepo: DataRepository ) {
+    protected dataRepo: DataRepository,
+    protected uprtclRepo: UprtclRepository ) {
   }
 
   async createData(
     data: Hashed<Object>, 
     _loggedUserId: string | null): Promise<string> {
 
     console.log('[UPRTCL-SERVICE] createData', data);
+
+    const dataCoded = {...data.object};
+    const castToObject = JSON.parse(JSON.stringify(dataCoded)); 
```

Comment:
Why do you need to both spread and parse? seems excessive. If the issue is with typescript complaining about the type of data.object, you can do a cast (data.object as any). Did you try that?

Reply:
That's correct!, thanks for the review. I'll add the fix.